### ISSUE TITLE
In JSpecify mode, missing meta-annotations trigger warning i.c.w. JDK 23+

### ISSUE THREAD

**Issue by Stephan202** [uber/NullAway#1171]
See [this discussion](https://github.com/PicnicSupermarket/error-prone-support/pull/1608#discussion_r2008789195) for context. I narrowed down the reproduction case to:
```java
import reactor.core.publisher.Mono;

final class Test {
  void m() {
    Mono.justOrEmpty(null);
  }
}
```

In the absence of a dependency on `com.google.code.findbugs:jsr305` (or similar `org.jspecify:jspecify` alternative), this triggers the following warning: 

```
[WARNING] Cannot find annotation method 'when()' in type 'javax.annotation.Nonnull': class file for javax.annotation.Nonnull not found
[WARNING] /path/to/Test.java: unknown enum constant javax.annotation.meta.When.MAYBE
  reason: class file for javax.annotation.meta.When not found
```

The trigger seems to be [this](https://github.com/reactor/reactor-core/blob/fe36085a17f8806b2906a87baf31a2e69af30e13/reactor-core/src/main/java/reactor/core/publisher/Mono.java#L764) usage of `reactor.util.annotation.Nullable`, which is [defined](https://github.com/reactor/reactor-core/blob/fe36085a17f8806b2906a87baf31a2e69af30e13/reactor-core/src/main/java/reactor/util/annotation/Nullable.java#L51-L52) with reference to a few `javax.annotation.*` meta annotations.

The issue is not observed without `-XepOpt:NullAway:JSpecifyMode=true`. I could also not reproduce it with JDK `21.0.6-tem` and `22.0.2-oracle`, but do observe it with `23.0.2-tem` and `24-open`. This indicates that it's specific to JSpecify mode on JDK 23+.

**Comment by msridhar** [comment]
I can reproduce this in a standalone build, but it's hard to reproduce in a unit test, so hard to track down exactly what is going on.  That said, what I think is going on is:

1. From JDK 23 on, more annotations get read from bytecodes via the Compiler Tree APIs, due to fixes in `javac`;
2. In JSpecify mode, we are calling one of these APIs on this method parameter that we didn't call before (due to some generics checking); and
3. reading the annotation via the Compiler Tree API leads to the annotation type being properly loaded by `javac`, which leads to its meta-annotations being loaded, which leads to the warning you are seeing.

The only part I'm not too sure about is why 2 is happening (which is hard to track down outside a unit test).  But I don't think anything wrong is going on here.  `javac` tries to be lazy about loading types / annotations from the classpath, but in principle it can load whatever is present there and whatever those types depend on.  By using `javax.annotations.*` meta annotations, I think those annotations become part of `reactor.util.annotation.Nullable`'s "API" and should be on the classpath to avoid such warnings.

While I haven't fully tracked this down, I'm pretty sure it's not a NullAway bug, so I'm going to close it out.  Unfortunately I think you'll need the JSR305 annotations on the classpath to avoid these warnings for now.
